### PR TITLE
Learner-3791 Android - Automate New Logistration screen

### PR DESCRIPTION
**Pull request by nedo**
ready for review

### CODE REVIEW

File: pages/new_logistration.py [hunk 1]
```diff
@@ -119,15 +123,141 @@ def load_register_screen(self):
...
+        if InputData.target_environment == 'iOS':
+            if self.load_login_screen().text == Strings.LOGIN_SCREEN_TITLE:
+                all_buttons = self.driver.find_elements_by_class_name(self.elements.all_buttons)
+                all_buttons[0].click()
+                sleep(self.global_contents.minimum_timeout)
+                if self.get_discover_course_button().text == Strings.NEW_LOGIS_DISCOVER_COURSES:
+                    return True
+                else:
+                    print("Problem getting back from Login")
+                    return False
```
Comment (staubina) [inline pages/new_logistration.py]:
```
                if self.get_discover_course_button().text == Strings.NEW_LOGIS_DISCOVER_COURSES:
                     return True
                 else:
                     print("Problem getting back from Login")
                     return False
```

can be 

```
return self.get_discover_course_button().text == Strings.NEW_LOGIS_DISCOVER_COURSES
```

File: pages/new_logistration.py [hunk 2]
```diff
@@ -119,15 +123,141 @@ def load_register_screen(self):
...
+            else:
+                print("Problem going to Login")
+                return False
+
+        elif InputData.target_environment == Strings.ANDROID:
+            if self.load_login_screen() == Globals.LOGIN_ACTIVITY_NAME:
+                self.driver.back()
+                sleep(self.global_contents.minimum_timeout)
+                if self.driver.current_activity == Globals.NEW_LOGISTRATION_ACTIVITY_NAME:
```
Comment (staubina) [inline pages/new_logistration.py]: Same comment as above, this logic can be simplified.

File: pages/new_logistration.py [hunk 3]
```diff
@@ -119,15 +123,141 @@ def load_register_screen(self):
...
+
+        Returns:
+             True or False
+        """
+
+        if InputData.target_environment == 'iOS':
+            if self.load_register_screen().text == Strings.REGISTER_SCREEN_TITLE:
+                all_buttons = self.driver.find_elements_by_class_name(self.elements.all_buttons)
+                all_buttons[0].click()
+                sleep(self.global_contents.minimum_timeout)
```
Comment (staubina) [inline pages/new_logistration.py]: You repeat the steps above multiple times in the tests, could they be moved to a method, click_login_button or something like that.

File: pages/new_logistration.py [hunk 4]
```diff
@@ -119,15 +123,141 @@ def load_register_screen(self):
...
+                if self.get_discover_course_button().text == Strings.NEW_LOGIS_DISCOVER_COURSES:
+                    return True
+                else:
+                    print("Problem getting back from Register")
+                    return False
```
Comment (staubina) [inline pages/new_logistration.py]: Simplify the above conditional.  It looks like this is a repeated example and can be simplified across the board.

File: pages/new_logistration.py [hunk 5]
```diff
@@ -119,15 +123,141 @@ def load_register_screen(self):
...
+            if self.load_register_screen() == Globals.REGISTER_ACTIVITY_NAME:
+                self.driver.back()
+                sleep(self.global_contents.minimum_timeout)
+                if self.driver.current_activity == Globals.NEW_LOGISTRATION_ACTIVITY_NAME:
+                    return True
+                else:
+                    print("Problem getting back from Register")
+                    return False
+            else:
+                print("Problem going to Register")
```
Comment (staubina) [inline pages/new_logistration.py]: Note: I am not sure these print statements are useful at they currently are.  The tests all print the same error.  In a future improvement, we should try to be more clear with error messaging.

File: pages/new_logistration.py [hunk 6]
```diff
@@ -119,15 +123,141 @@ def load_register_screen(self):
...
+                return False
+
+    def back_and_forth_dicover_courses(self):
+        """
+        Load register screen and get back to previous screen
+
+        Returns:
+             True or False
```
Comment (staubina) [inline pages/new_logistration.py]: You returns comments should explain what the false or true means, not just that it returns a boolean.

consider the following

```
bool: Returns True if <Insert statement>
```

File: pages/new_logistration.py [hunk 7]
```diff
@@ -119,15 +123,141 @@ def load_register_screen(self):
...
+        """
 
-        return self.driver.current_activity
\ No newline at end of file
+        if InputData.target_environment == 'iOS':
```
Comment (staubina) [inline pages/new_logistration.py]: Please use the constant

File: tests/test_new_logistration.py [hunk 8]
```diff
@@ -50,4 +55,23 @@ def test_ui_elements_smoke(self, set_capabilities):
...
+                    back to 'New Logistration' screen.
+                Verify tapping "Register" loads Register screen
+                Verify tapping back icon from 'Register' screen navigate user
+                    back to 'New Logistration' screen. 
+                Verify tapping "Discover Courses" loads Discovery screen
+                Verify tapping back icon from 'Discover Courses' screen
+                    navigate user back to 'New Logistration' screen.
+        """
+        new_logistration_page = NewLogistration(set_capabilities)
+        assert new_logistration_page.back_and_forth_login() is not False
```
Comment (staubina) [inline tests/test_new_logistration.py]: This would be better if it was done in the positive.

```
assert new_logistration_page.back_and_forth_login()
```

**Comment by staubina** [comment]
Also @MirzaMubasharBaig we should have someone review this PR for Android Workflows to ensure we are testing as expected.

File: pages/new_logistration.py [hunk 9]
```diff
@@ -119,15 +123,141 @@ def load_register_screen(self):
...
+    def load_discover_courses_screen(self):
+        """
+        Load Discover Courses Screen
+
+        Returns:
+             If Android - Discover Courses Activity Name
+             If iOS - Discover Courses screen Title Name
+        """
+        self.get_discover_course_button().click()
+        sleep(self.global_contents.medium_timeout)
```
[the remark this answers is not in the document]
Comment (nedo) [reply pages/new_logistration.py]: Created a ticket for this behavior, 

LEARNER-3866 - Implement a global function to wait for availability of element

**Comment by nedo** [comment]
All highlighted points are addressed, please have a look and comment for further.

File: common/globals.py [hunk 10]
```diff
@@ -45,7 +43,7 @@ def validate_element(self, target_element, element_value, expected_value, error_
             target_element, element_value, expected_value, error_msg
 
         Return:
-            element
+            Element
```
Comment (staubina) [inline common/globals.py]: In the future we should to add more detail to the return values describing their type and what they are.

File: pages/login.py [hunk 11]
```diff
@@ -48,14 +48,16 @@ def get_title_textview(self):
...
-            return self.global_contents.validate_element(textview_screen_title,
-                                                         textview_screen_title.text,
-                                                         Strings.LOGIN_SCREEN_TITLE,
-                                                         Strings.ERROR)
+            return self.global_contents.validate_element(
+                textview_screen_title,
+                textview_screen_title.text,
+                Strings.LOGIN_SCREEN_TITLE,
+                Strings.ERROR
+            )
```
Comment (staubina) [inline pages/login.py]: 👍

File: pages/new_logistration.py [hunk 12]
```diff
@@ -158,106 +168,85 @@ def load_discover_courses_screen(self):
...
                 sleep(self.global_contents.minimum_timeout)
-                if self.get_discover_course_button().text == Strings.NEW_LOGIS_DISCOVER_COURSES:
-                    return True
-                else:
-                    print("Problem getting back from Login")
-                    return False
+                return self.get_discover_course_button().text == Strings.NEW_LOGIS_DISCOVER_COURSES:
             else:
-                print("Problem going to Login")
+            print("Problem - Login screen is not loaded ")
```
Comment (staubina) [inline pages/new_logistration.py]: Is the spacing off here?

File: pages/new_logistration.py [hunk 13]
```diff
@@ -158,106 +168,85 @@ def load_discover_courses_screen(self):
...
                 sleep(self.global_contents.minimum_timeout)
-                if self.driver.current_activity == Globals.NEW_LOGISTRATION_ACTIVITY_NAME:
-                    return True
-                else:
-                    print("Problem getting back from Login")
-                    return False
+                return self.driver.current_activity == Globals.NEW_LOGISTRATION_ACTIVITY_NAME:
             else:
-                print("Problem going to Login")
+            print("Problem - Login screen is not loaded ")
```
Comment (staubina) [inline pages/new_logistration.py]: The spacing looks off on this line also.

File: common/globals.py (hunk 10 above)
Comment (nedo) [reply common/globals.py]: Basically it returns specific element whose type is 'webdriver element', I've changed it. Please have a look and guide to improve it more better, 

'''
Returns:
            webdriver element: Login Button element
'''

**Pull request merged**